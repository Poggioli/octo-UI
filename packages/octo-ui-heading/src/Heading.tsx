import { CSS, forwardRef } from "@octo-ui/core";
import { StyledHeading } from "./styles";
import { DEFAULT_TAG_HEADING, HeadingProps } from "./types";

const Heading = forwardRef<typeof StyledHeading, HeadingProps>(
  ({ color, ...props }, forwardedRef) => {
    const css: CSS = {
      color: color ?? "$grey12",
      ...props.css,
    };

    return (
      <StyledHeading
        as={DEFAULT_TAG_HEADING}
        {...props}
        ref={forwardedRef}
        size={{
          "@initial": "6",
          "@medium": "7",
        }}
        css={{
          fontWeight: 500,
          fontVariantNumeric: "proportional-nums",
          lineHeight: "25px",
          ...css,

          "@medium": {
            lineHeight: "30px",
            ...css["@medium"],
          },
        }}
      />
    );
  }
);
Heading.toString = () => `.${StyledHeading.className}`;

export { StyledHeading };
export { Heading };
