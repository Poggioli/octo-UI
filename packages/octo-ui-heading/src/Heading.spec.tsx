import { render } from "@testing-library/react";
import { Heading } from "./Heading";

describe("<Heading />", () => {
  it(`GIVEN a <Heading />
      WHEN screen is small than 900px
      THEN should render a <Heading /> using size 3
      AND lineHeight 25px`, () => {
    global.innerWidth = 768;
    const rendered = render(<Heading>Heading</Heading>);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Heading />
      WHEN screen is equal or greater than 900px
      THEN should render a <Heading /> using size 4
      AND lineHeight 27px`, () => {
    global.innerWidth = 900;
    const rendered = render(<Heading>Heading</Heading>);
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Heading />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Heading.toString()).toBe(".octo-UI-c-gWerhV");
  });

  describe("color props", () => {
    it(`GIVEN a <Heading />
        WHEN color props is not setted
        THEN should render a <Heading /> $grey11 as color`, () => {
      const rendered = render(<Heading>Heading</Heading>);
      expect(rendered.container).toMatchSnapshot();
    });

    it(`GIVEN a <Heading />
        WHEN color props is $grey10
        THEN should render a <Heading /> $grey10 as color`, () => {
      const rendered = render(<Heading color="$grey10">Heading</Heading>);
      expect(rendered.container).toMatchSnapshot();
    });
  });
});
