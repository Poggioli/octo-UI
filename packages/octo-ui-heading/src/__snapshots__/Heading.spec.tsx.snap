// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> GIVEN a <Heading />
      WHEN screen is equal or greater than 900px
      THEN should render a <Heading /> using size 4
      AND lineHeight 27px 1`] = `
<div>
  <h1
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-cMAlre-size-6 octo-UI-c-gWerhV-cPVmnY-size-7 octo-UI-c-gWerhV-ikZhbNd-css"
  >
    Heading
  </h1>
</div>
`;

exports[`<Heading /> GIVEN a <Heading />
      WHEN screen is small than 900px
      THEN should render a <Heading /> using size 3
      AND lineHeight 25px 1`] = `
<div>
  <h1
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-cMAlre-size-6 octo-UI-c-gWerhV-cPVmnY-size-7 octo-UI-c-gWerhV-ikZhbNd-css"
  >
    Heading
  </h1>
</div>
`;

exports[`<Heading /> color props GIVEN a <Heading />
        WHEN color props is $grey10
        THEN should render a <Heading /> $grey10 as color 1`] = `
<div>
  <h1
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-cMAlre-size-6 octo-UI-c-gWerhV-cPVmnY-size-7 octo-UI-c-gWerhV-ijRaJtX-css"
  >
    Heading
  </h1>
</div>
`;

exports[`<Heading /> color props GIVEN a <Heading />
        WHEN color props is not setted
        THEN should render a <Heading /> $grey11 as color 1`] = `
<div>
  <h1
    class="octo-UI-c-gWerhV octo-UI-c-PJLV octo-UI-c-gWerhV-cMAlre-size-6 octo-UI-c-gWerhV-cPVmnY-size-7 octo-UI-c-gWerhV-ikZhbNd-css"
  >
    Heading
  </h1>
</div>
`;
